### PR TITLE
Add OWASP dependency check issue parser

### PR DESCRIPTION
**Pull request by sirius94**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

This PR is a continuation of https://github.com/jenkinsci/analysis-model/pull/708 and depends on https://github.com/jenkinsci/warnings-ng-plugin/pull/1115

### CODE REVIEW

**Comment by KalleOlaviNiemitalo** [comment]
I assume this does not have the same SECURITY-2488 XXE vulnerability as the [OWASP Dependency-Check](https://plugins.jenkins.io/dependency-check-jenkins-plugin/) plugin.

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/OwaspDependencyCheck.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    /** Create instance. */
+    @DataBoundConstructor
+    public OwaspDependencyCheck() {
+        super();
+        // empty constructor required for stapler
+    }
+
+    /** Descriptor for this static analysis tool. */
+    @Symbol("owaspDependencyCheck")
```
Comment (uhafner) [inline plugin/src/main/java/io/jenkins/plugins/analysis/warnings/OwaspDependencyCheck.java:25]: Did you check if this ID is not already used by the original OwaspDependencyCheck plugin?

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
> I assume this does not have the same SECURITY-2488 XXE vulnerability as the [OWASP Dependency-Check](https://plugins.jenkins.io/dependency-check-jenkins-plugin/) plugin.

Actually the parser uses JSON. (And all parsers in the warnings plugin correctly use a safe parser: https://github.com/jenkinsci/analysis-model/blob/master/src/main/java/edu/hm/hafner/analysis/SecureXmlParserFactory.java)

File: plugin/src/main/java/io/jenkins/plugins/analysis/warnings/OwaspDependencyCheck.java (hunk 1 above)
Comment (KalleOlaviNiemitalo) [reply plugin/src/main/java/io/jenkins/plugins/analysis/warnings/OwaspDependencyCheck.java:25]: The original has:

- [`@Symbol("dependency-check")`](https://github.com/jenkinsci/dependency-check-plugin/blob/dependency-check-jenkins-plugin-5.1.1/src/main/java/org/jenkinsci/plugins/DependencyCheck/tools/DependencyCheckInstallation.java#L93)
- [`@Symbol("dependencyCheckPublisher")`](https://github.com/jenkinsci/dependency-check-plugin/blob/dependency-check-jenkins-plugin-5.1.1/src/main/java/org/jenkinsci/plugins/DependencyCheck/DependencyCheckPublisher.java#L173)
- [`@Symbol({"dependencyCheck", "dependencycheck"})`](https://github.com/jenkinsci/dependency-check-plugin/blob/dependency-check-jenkins-plugin-5.1.1/src/main/java/org/jenkinsci/plugins/DependencyCheck/DependencyCheckToolBuilder.java#L220)

[Pipeline Step Reference](https://www.jenkins.io/doc/pipeline/steps/dependency-check-jenkins-plugin/) doesn't list any others for that plugin, either.

**Comment by uhafner** [comment]
Do not use force push after we made a review. Otherwise it is hard to see what changed since the review.

**Pull request merged**